src/runtime_src/driver/xclng/tools/xbutil/xbutil.h

```diff
@@ -959,20 +963,21 @@ class device {
         ostr << "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n";
         ostr << "Compute Unit Status\n";
         try {
+          int cu_i = 0;
           for (auto& v : sensor_tree::get_child( "board.compute_unit" )) {
             int index = std::stoi(v.first);
             if( index >= 0 ) {
               std::string cu_n, cu_s, cu_ba;
-              int cu_i = 0;
               for (auto& subv : v.second) {
                 if( subv.first == "name" )
                   cu_n = subv.second.get_value<std::string>();
-                else if( subv.first == "base_address" )
-                  cu_ba = sensor_tree::pretty<int>(subv.second.get_value<int>(), "N/A", true);
-                else if( subv.first == "status" )
+                else if( subv.first == "base_address" ) {
+                  auto addr = subv.second.get_value<uint64_t>();
+		  cu_ba = (addr == (uint64_t)-1) ? "N/A" : sensor_tree::pretty<uint64_t>(addr, "N/A", true);
+		} else if( subv.first == "status" )
                   cu_s = subv.second.get_value<std::string>();
               }
-              ostr << "CU[" << std::right << std::setw(2) << cu_i << "]: "
+              ostr << "CU[" << std::right << std::setw(2) << cu_i++ << "]: "
```

Comment:
Having this `cu_i++` lost in the middle of an output statement makes the code difficult to its own line?

Reply:
did this. Thanks for your comment.